src/core/shutdown.c

```diff
@@ -158,6 +158,41 @@ static int switch_root_initramfs(void) {
         return switch_root("/run/initramfs", "/oldroot", false, MS_BIND);
 }
 
+static int sync_with_timeout(void) {
+        sigset_t mask, oldmask;
+        pid_t pid;
+        int r;
+
+        log_info("Syncing filesystems.");
+
+        assert_se(sigemptyset(&mask) == 0);
+        assert_se(sigaddset(&mask, SIGCHLD) == 0);
+        assert_se(sigprocmask(SIG_BLOCK, &mask, &oldmask) == 0);
+
+        /* Due to the possiblity of a remount operation hanging, we
```

Comment:
s/remount/sync/

Reply:
and a couple more below

Reply:
Ugh... I don't know how that slipped past. I will correct it with a further push. Since github masks comments after a force push, I will hold off for a bit more feedback first.